brainiak/reprsimil/brsa.py

```diff
@@ -1659,3 +2460,1350 @@ def _loglike_AR1_singpara(self, param, XTX, XTDX, XTFX, YTY_diag,
+        we assume that a set of time courses not related to the task
+        conditions are shared across voxels with unknown amplitudes.
+        One approach is for users to provide time series which they consider
+        as nuisance but exist in the noise (such as head motion).
+        The other way is to take the first n_nureg principal components
+        in the residual after subtracting the response to the design matrix
+        from the data, and use these components as the nuisance regressor.
+        This flag is for the second approach. If turned on,
+        PCA or factor analysis will be applied to the residuals
+        to obtain new nuisance regressors in each round of fitting.
+        These two approaches can be combined. If the users provide nuisance
+        regressors and set this flag as True, then the first few principals
+        of the residuals after subtracting both the responses to design
+        matrix and nuisance regressors will be used in addition to the
+        nuisance regressors provided by the users.
+        Note that nuisance regressor is not required from user. If it is
+        not provided, DC components for each run will be included as nuisance
+        regressor regardless of the auto_nuisance parameter.
+    n_nureg: int or string. Default: 'BIC'
+        Number of nuisance regressors to use in order to model signals
+        shared across voxels not captured by the design matrix.
+        This number is in addition to any nuisance regressor that the user
+        has already provided.
+        If set to 'BIC', the number of nuisance regressor will be
+        automatically determined based on BIC metric
+        of probabalistic PCA model. (Minka 2000, NIPS)
+    nureg_method: string, 'PCA', 'FA', 'ICA' or 'SPCA'. Default: 'ICA'
+        The method to estimate the shared component in noise across voxels.
+    DC_single: boolean. Default: True
+        A time course of constant 1 will be included to the nuisance
+        regressor for each participant. If DC_single is set to False,
+        one such regressor is included for each fMRI run, but at the end of
+        fitting, a single component in beta0\_ will be computed as the average
+        of the weight maps corresponding to these regressors. This might
+        cause underestimation of noise variance.
+        If DC_single is True, only one regressor of constant 1 will be
+        used for the whole dataset. This might be desirable if you
+        believe the average image intensity might not scale with the
+        same proportion for different voxels across scan. In other words,
+        it is possible that some part of the brain is more vulnerable to
+        change in baseline intensity due to facts such as
+        field inhomogeneity. Setting DC_single to True will force the
+        nuisance regressors automatically estimated from residuals to
+        capture this.
+    logS_range: float. Default: 1.0
```

Comment:
It has been made clear to be truncated distribution in the SNR_prior argument now. For the second question, log normal distribution here assumes the mean on log scale is 0 (1 in linear scale). The actual mean does not matter because `L_` will scale accordingly. I think it is reasonable to use symmetric range on both sides on the log scale.